components/org.wso2.carbon.identity.application.authenticator.biometric.device.handler/src/main/java/org/wso2/carbon/identity/application/authenticator/biometric/device/handler/impl/DeviceHandlerImpl.java

```diff
@@ -0,0 +1,155 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.biometric.device.handler.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.DeviceHandler;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.cache.BiometricDeviceHandlerCacheKey;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.cache.DeviceCache;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.cache.DeviceCacheEntry;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.cache.RegistrationRequestChallengeCache;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.cache.RegistrationRequestChallengeCacheEntry;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.dao.DeviceDAOImpl;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.exception.BiometricDeviceHandlerClientException;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.exception.BiometricdeviceHandlerServerException;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.model.Device;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.model.DiscoveryData;
+import org.wso2.carbon.identity.application.authenticator.biometric.device.handler.model.RegistrationRequest;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.io.IOException;
+
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.UUID;
+
+/**
+ * This class implements the DeviceHandler interface .
+ */
+public class DeviceHandlerImpl implements DeviceHandler {
+    private static final Log log = LogFactory.getLog(DeviceHandler.class);
+
+    @Override
+    public Device registerDevice(RegistrationRequest registrationRequest)
+            throws BiometricDeviceHandlerClientException, BiometricdeviceHandlerServerException, SQLException,
+            UserStoreException, JsonProcessingException, NoSuchAlgorithmException,
+            SignatureException, InvalidKeySpecException, InvalidKeyException {
```

Comment:
Add a new line after the method signature. Fix this all other places applicable.